supervisor/security/module.py

```diff
@@ -0,0 +1,169 @@
+"""Fetch last versions from webserver."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from ..const import (
+    ATTR_CONTENT_TRUST,
+    ATTR_FORCE_SECURITY,
+    ATTR_PWNED,
+    FILE_HASSIO_SECURITY,
+)
+from ..coresys import CoreSys, CoreSysAttributes
+from ..exceptions import (
+    CodeNotaryError,
+    CodeNotaryUntrusted,
+    PwnedError,
+    SecurityJobError,
+)
+from ..jobs.decorator import Job, JobCondition, JobExecutionLimit
+from ..resolution.const import ContextType, IssueType
+from ..utils.codenotary import cas_validate
+from ..utils.common import FileConfiguration
+from ..utils.pwned import check_pwned_password
+from ..validate import SCHEMA_SECURITY_CONFIG
+from .const import ContentTrustResult, IntegrityResult
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class Security(FileConfiguration, CoreSysAttributes):
+    """Handle Security properties."""
+
+    def __init__(self, coresys: CoreSys):
+        """Initialize updater."""
+        super().__init__(FILE_HASSIO_SECURITY, SCHEMA_SECURITY_CONFIG)
+        self.coresys = coresys
+
+    @property
+    def content_trust(self) -> bool:
+        """Return if content trust is enabled/disabled."""
+        return self._data[ATTR_CONTENT_TRUST]
+
+    @content_trust.setter
+    def content_trust(self, value: bool) -> None:
+        """Set content trust is enabled/disabled."""
+        self._data[ATTR_CONTENT_TRUST] = value
+
+    @property
+    def force(self) -> bool:
+        """Return if force security is enabled/disabled."""
+        return self._data[ATTR_FORCE_SECURITY]
+
+    @force.setter
+    def force(self, value: bool) -> None:
+        """Set force security is enabled/disabled."""
+        self._data[ATTR_FORCE_SECURITY] = value
+
+    @property
+    def pwned(self) -> bool:
+        """Return if pwned is enabled/disabled."""
+        return self._data[ATTR_PWNED]
+
+    @pwned.setter
+    def pwned(self, value: bool) -> None:
+        """Set pwned is enabled/disabled."""
+        self._data[ATTR_PWNED] = value
+
+    async def verify_content(self, signer: str, checksum: str) -> None:
+        """Verify content on CAS."""
+        if not self.content_trust:
+            _LOGGER.warning("Disabled content-trust, skip validation")
+            return
+
+        try:
+            await cas_validate(signer, checksum)
+        except CodeNotaryUntrusted:
+            raise
+        except CodeNotaryError:
+            if self.force:
+                raise
+            return
```

Comment:
@pvizeli after this change this is now the only way untrusted software could actually get installed and stick around. Very few if any users will run with `force=true` so on a CAS issue we'll just install the software and move on. Previously we'd evaluate it again later but now that won't happen.

Should we put in some way to follow-up on this and check again later in this situation? It'll be too late to stop the install at that point but we could inform the user with an issue. Maybe we leave the trust evaluation but its exclusively used for following up on CAS issues and not for regular trust checks of already checked software?

Reply:
Oh that is a great idea. Create a suggestion to run the integrity check if it fails. I would suggest adding this on a new PR, after the release is out. I don't want block the release but also not having such implementation only 1-2 days on beta before we have to move into stable. This make it at least better as we have on stable but yeah, we have to optimize it